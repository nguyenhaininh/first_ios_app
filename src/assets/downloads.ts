import { BookData } from './books';

export interface DownloadData {
  book: BookData;
  deliveryDate: string;
  endDate: string;
  size: number;
  downloaded: number;
}

const downloads: DownloadData[] = [
  {
    book: {
      title: 'mina 7月号',
      date: '2018/5/15',
      image: require('./images/dummy/cover-1.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: 'non-no 5月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-3.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  }, {
    book: {
      title: 'GISELe 6月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-33.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: '&Premium 7月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-15.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  }, {
    book: {
      title: 'Hanako 6月12日号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-13.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 84.74,
    downloaded: 100,
  }, {
    book: {
      title: 'FRaU 7月号',
      date: '2018.5.15',
      image: require('./images/dummy/cover-20.png')
    },
    deliveryDate: '2018/5/20',
    endDate: '2018/8/19',
    size: 20.21,
    downloaded: 30,
  },
];

export default downloads;
