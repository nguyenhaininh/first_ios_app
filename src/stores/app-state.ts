/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
export interface IAppState {
  informations?: IAppInformation[];
  isShowingMenu: boolean;
  mode: AppMode;
  tab: AppTab;
}

export enum AppMode {
  Init,
  Login,
  Member,
}

export enum AppTab {
  Feed,
  Ranking,
  Favorite,
  Download,
  Search
}

export interface IAppInformation {
  infoId: string;
  subject: string;
  description: string;
  date: string;
  important?: boolean;
}

export function createInitialAppState(): IAppState {
  return {
    isShowingMenu: false,
    mode: AppMode.Init,
    tab: AppTab.Feed,
  };
}
