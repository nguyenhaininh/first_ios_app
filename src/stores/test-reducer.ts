/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import { incrementAction, decrementAction, resetAction } from '../actions/test-actions';

export interface ITestState {
  counter: number;
  message?: string;
}

export function createInitialTestState(): ITestState {
  return { counter: 0 };
}

export const incrementReducer = incrementAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
    };
  }
  return {
    message: undefined,
    counter: state.counter += action.payload,
  };
});

export const decrementReducer = decrementAction.reducer<ITestState>((state, action) => ({
  counter: state.counter -= action.payload,
}));

export const resetReducer = resetAction.reducer<ITestState>((state, action) => {
  if (isError(action)) {
    return {
      message: action.payload && action.payload.message,
      counter: 0
    };
  }
  return {
    message: undefined,
    counter: 0
  };
});

export const testReducer = createTypeReducer(
  createInitialTestState,
  incrementReducer,
  decrementReducer,
  resetReducer,
);
