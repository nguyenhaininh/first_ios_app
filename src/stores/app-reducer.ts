/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createTypeReducer, isError } from '../common/utility/redux';
import {
  initAppAction, fetchInfomartionsAction, setShowAppMenu, setTabAction, setModeAction,
} from '../actions/app-actions';
import { IAppState, createInitialAppState, AppMode } from './app-state';

export const appReducer = createTypeReducer<IAppState>(
  createInitialAppState,
  initAppAction.reducer((state, action) => {
    if (isError(action)) {
      throw action.error;
    }
    return {
      ...state,
      mode: AppMode.Login,
    };
  }),
  fetchInfomartionsAction.reducer((state, action) => isError(action) ? ({}) : ({
    informations: action.payload
  })),
  setShowAppMenu.reducer((state, action) => ({
    isShowingMenu: action.payload
  })),
  setTabAction.reducer((state, action) => ({
    tab: action.payload
  })),
  setModeAction.reducer((state, action) => ({
    mode: action.payload
  })),
);
