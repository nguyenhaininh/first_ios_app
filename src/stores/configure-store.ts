/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import { createStore, combineReducers, applyMiddleware } from 'redux';

import { typePendingReducerSet, TypeReduxPendingState, createTypeReduxInitialState, typeReduxMiddleware } from '../common/utility/redux';

import { ITestState, testReducer, createInitialTestState } from './test-reducer';
import { appReducer } from './app-reducer';
import { createInitialAppState, IAppState } from './app-state';

export interface IStore extends TypeReduxPendingState {
  testState: ITestState;
  appState: IAppState;
}

export const rootReducer = combineReducers<IStore>({
  ...typePendingReducerSet,
  testState: testReducer,
  appState: appReducer,
});

const middlewares = [typeReduxMiddleware];

export const InitialState: IStore = {
  ...createTypeReduxInitialState(),
  testState: createInitialTestState(),
  appState: createInitialAppState(),
};

export const configureStore = (initialState: IStore = InitialState) => createStore(rootReducer, initialState, applyMiddleware(...middlewares));
