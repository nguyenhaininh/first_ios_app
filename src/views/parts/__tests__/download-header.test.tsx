/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import { Provider } from 'react-redux';
import DownloadHeader from '../download-header';

import * as renderer from 'react-test-renderer';
import { configureStore } from '../../../stores/configure-store';

import downloads from '../../../assets/downloads';

const onPress = () => {};

it('renders correctly', () => {
  const store = configureStore();
  const tree = renderer.create(
    <Provider store={store}>
      <DownloadHeader order="order" used="used" total="total" />
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
