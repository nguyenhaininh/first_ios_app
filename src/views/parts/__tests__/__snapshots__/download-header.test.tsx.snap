// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f7f7f7",
      "borderBottomColor": "#eaebed",
      "borderBottomWidth": 0.5,
      "height": 111,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingVertical": 9,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4f5b77",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        [missing "en.device_capacity" translation]
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4f5b77",
            "fontSize": 12,
          }
        }
      >
        [missing "en.already_used" translation]: used/total
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 22,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff200",
            },
            Object {
              "flex": 1,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e0e0e0",
            },
            Object {
              "flex": 3,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingVertical": 9,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/delete_ico.png",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#4f5b77",
              "fontSize": 12,
              "marginLeft": 5,
            }
          }
        >
          [missing "en.delete_all" translation]
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 5,
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingVertical": 10,
          "position": "absolute",
          "right": 15,
          "width": 100,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/sort_ico.png",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2f3f62",
            "fontSize": 12,
            "marginLeft": 8,
          }
        }
      >
        order
      </Text>
    </View>
  </View>
</View>
`;
